app/src/main/java/com/codesoom/domain/ProductRepository.java

```diff
@@ -0,0 +1,89 @@
+package com.codesoom.domain;
+
+import com.codesoom.exception.ProductNotFoundException;
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * 상품 데이터를 관리합니다.
+ */
+
+@Repository
+public class ProductRepository {
+    private final Map<Long, Product> products = new LinkedHashMap<>();
+    private Long newId = 0L;
+
+    /**
+     * 상품 목록을 반환해준다.
+     *
+     * @return 상품 목록
+     */
+    public List<Product> findAll() {
+        return new ArrayList<>(products.values());
+    }
+
+    /**
+     * id에 해당하는 상품을 반환해준다.
+     *
+     * @param id 상품 id
+     * @return id에 해당하는 상품
+     */
+    public Product find(Long id) {
+       Product product = findProduct(id);
+
+        return product;
+    }
+
+    /**
+     * 상품의 정보를 저장하고 반환한다.
+     *
+     * @param product 저장할 상품
+     * @return 저장한 상품
+     */
+    public Product save(Product product) {
+        Long id = generateId();
+        product.setId(id);
+        products.put(id, product);
+
+        return product;
+    }
+
+    /**
+     * 상품을 삭제한다.
+     *
+     * @param product 삭제할 상품
+     */
+    public void remove(Product product) {
+        products.remove(product.getId());
+    }
+
+    /**
+     * id에 해당하는 상품이 있다면 상품을 리턴하고 없다면 찾을 수 없다는 예외를 던진다.
+     *
+     * @param id 상품 아이디
+     * @return id에 해당하는 상품
+     * @exception ProductNotFoundException
```

Comment:
`@exception`은 이제 사용하지 않습니다. `@throws`를 사용해 주세요. `@throws`는 예외 타입, 예외를 어떤 경우에 던지는지의 순서로 작성해줍니다.
```suggestion
     * @throws ProductNotFoundException 상품을 찾을 수 없는 경우
```